### DISCUSSION TITLE
Unable to install Ruby 3.2.0

### DISCUSSION THREAD

**Discussion by mirnicscsongor** [rbenv/ruby-build#2118] [Build failures] [answered]
So I run 
```
rbenv install 3.2.0
rbenv global 3.2.0
```
and this is the error I get
```
To follow progress, use 'tail -f /var/folders/59/dgs3jq4n2fn03zlts84hnryc0000gn/T/ruby-build.20221227153230.4209.log' or pass --verbose
Downloading openssl-3.0.7.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/83049d042a260e696f62406ac5c08bf706fd84383f945cf21bd61e9ed95c396e
Installing openssl-3.0.7...
Installed openssl-3.0.7 to /Users/csongormirnics/.rbenv/versions/3.2.0

Downloading ruby-3.2.0.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.2/ruby-3.2.0.tar.gz
Installing ruby-3.2.0...
ruby-build: using readline from homebrew

BUILD FAILED (macOS 13.1 using ruby-build 20221225)

Inspect or clean up the working tree at /var/folders/59/dgs3jq4n2fn03zlts84hnryc0000gn/T/ruby-build.20221227153230.4209.n1dfxN
Results logged to /var/folders/59/dgs3jq4n2fn03zlts84hnryc0000gn/T/ruby-build.20221227153230.4209.log

Last 10 log lines:
	Check ext/psych/mkmf.log for more details.
*** Fix the problems, then remove these directories and try again if you want.
Generating RDoc documentation
/private/var/folders/59/dgs3jq4n2fn03zlts84hnryc0000gn/T/ruby-build.20221227153230.4209.n1dfxN/ruby-3.2.0/lib/yaml.rb:3: warning: It seems your ruby installation is missing psych (for YAML output).
To eliminate this warning, please install libyaml and reinstall your ruby.
uh-oh! RDoc had a problem:
cannot load such file -- psych

run with --debug for full backtrace
make: *** [rdoc] Error 1
rbenv: version `3.2.0' not installed
```

**Comment by hsbt** [comment, marked as answer]
>To eliminate this warning, please install libyaml and reinstall your ruby.

You should install libyaml libraries and development header from package system like homebrew.

**Reply by chrise86** [reply]
I'm also facing this issue but have `libyaml` installed already. When attempting to install `libyaml-devel` as mentioned elsewhere (although for Linux, I believe) Brew just redirects to the `libyaml` package and says it's already installed. Any ideas?

**Reply by vkononov** [reply]
I can confirm that on my Mac OS 13.1, I am able to install Ruby 3.2.0 while only having `libyaml` installed using brew. Do you have an older OS? That could definitely be the culprit.

**Reply by chrise86** [reply]
Yep, fully up to date (13.1) brand new Mac Studio.

**Reply by chrise86** [reply]
Managed to fix using [this answer](https://github.com/asdf-vm/asdf-ruby/issues/198#issuecomment-1216901248), although unsure why it was needed.

**Reply by aflansburg** [reply]
> Managed to fix using [this answer](https://github.com/asdf-vm/asdf-ruby/issues/198#issuecomment-1216901248), although unsure why it was needed.

Thanks @chrise86 

M1 MBP on 13.1 Here - I had to do the same after finding that `libyaml-devel` was not available. Crucial note, I had to set:
```
export CFLAGS="-Wno-error=implicit-function-declaration"
```
in order for this to work. The reason for that appeared to be related to compilers finding `libffi` which was apparent in this message:
```
libffi is keg-only, which means it was not symlinked into /opt/homebrew,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

For compilers to find libffi you may need to set:
  export LDFLAGS="-L/opt/homebrew/opt/libffi/lib"
  export CPPFLAGS="-I/opt/homebrew/opt/libffi/include"
```

**Reply by rockotseng** [reply]
`brew update && brew upgrade ruby-build` this solves the issue at my end.

**Comment by alemens** [comment]
ruby-3.2.0/lib/yaml.rb `require` 'psych' so `rbenv install 3.2.0` without `libyaml` is a failure 

instead `rbenv install 3.1.3` is a success

Environment 
```
rbenv 1.2.0
Homebrew 3.6.16-97-ge76c55e
zsh 5.8.1 (x86_64-apple-darwin22.0)
macOS Ventura Version 13.1 (22C65) x86_64
```

**Reply by dcts** [reply]
how can you install libaml and psych?

**Comment by vkononov** [comment]
I had the same issue on CentOS, and running `sudo dnf install libyaml-devel` resolved the problem. `libyaml` alone was not enough. Needed to install `libyaml-devel`.

**Reply by baim-a** [reply]
thanks

**Reply by bl4ckouter** [reply]
Dude, I've been struggling with this for so long.
Thank you!

**Comment by joalbertg** [comment]
Steps I followed:
- `rbenv install --list` # not listed
- `brew install libyaml-devel libyaml`
- `brew update && brew upgrade ruby-build`
- `rbenv install --list` # Done!

**Reply by joelGarcia93** [reply]
you are the boss man!

**Reply by shileshv2** [reply]
I was installing ruby 3.x and got the error `error: Can't build Ruby: make failed: . ./vm_opts.h` 

`brew install libyml`
`frum install 3.2.2`
 
Ruby was installed successfully after installing libyml,

**Comment by kcheart** [comment]
Without installing libyaml, I can not install 3.2.0.

steps work for me (I did not install libyaml-devel)
- `brew install libyaml`
- `rbenv install 3.2.0`

environment:
- rbenv 1.2.0
- Homebrew 3.6.17
- zsh 5.8.1 (x86_64-apple-darwin22.0)
- macOS Ventura Version macOS 13.1 (22C65) M1

**Reply by snfxgsx** [reply]
hey, how did you manage to  install libyaml-devel ?

**Reply by vkononov** [reply]
He meant to say "Did *not* install libyaml-devel."

**Reply by kcheart** [reply]
sorry, my typo.
I did not install libyaml-devel.

**Comment by dcts** [comment]
I had the same issue, maybe useful to someone: Add psych to your gemfile, and specify the version to be below 4, because v4 has a bug.

Add to gemfile:
```rb
gem 'psych', '< 4'
```

Then run `bundle install`. That fixed it for me.

**Reply by kurko** [reply]
This seems unrelated to Ruby installation

**Comment by yaroslavrick** [comment]
In fedora I just need to update ruby-build:
`cd ~/.rbenv/plugins/ruby-build`
`git pull`
`rbenv install --list`
`rbenv install 3.2.0`

**Comment by yersel500** [comment]
Running this command before the installation of ruby 3.2.0 worked for me in Ubuntu 22.04:
```sudo apt install git curl libssl-dev libreadline-dev zlib1g-dev autoconf bison build-essential libyaml-dev libreadline-dev libncurses5-dev libffi-dev libgdbm-dev```

**Reply by gmcamposano** [reply]
Using Mint 21.1 can confirm that this worked. I just installed 3.2.1.

**Reply by edwardlayaodev** [reply]
thank you for this, having an issue on Pop OS 22.04, and this fixed it

**Comment by mathieujobin** [comment]
kept failing as well for me, it was hard to look at the log, because its running on docker build.
using ubuntu 20.04, installing `libyaml-dev` fixed the issue with 3.2.x

**Comment by nikos83** [comment]
On Catalina 
- brew install libyaml
- rbenv install 3.2.1
worked! Thanks!

**Comment by tnsasse** [comment]
If you run into this problem on Red Hat Enterprise Linux 9, you have to install the following packages:

- `dnf install libyaml`
-  `rpm -i https://mirror.stream.centos.org/9-stream/CRB/x86_64/os/Packages/libyaml-devel-0.2.5-7.el9.x86_64.rpm`

Note this will install `libyaml-devel` from the CentOS Stream 9 repository, as it is not available on RHEL.

**Comment by ccastillop** [comment]
M2 with Sonoma, 
The same error with libyaml and pyshc, which I've already installed with brew

This worked for me: https://github.com/TaKO8Ki/frum/issues/126#issuecomment-1520174616

**Comment by mathieujobin** [comment]
How about 3.2.2 ? there is no reason to install 3.2.0 anymore ?
